### ISSUE TITLE
Use distance for weight and delay in 3D space

### ISSUE THREAD

**Issue by dsc1126** [NeuralEnsemble/PyNN#514]
Hi there,

I am new to PyNN. I have built a Grid3D neuron network and specified space with periodic boundary on each side. I am trying to create projection to itself with 'AllToAllConnector'. I only want to assign weight if two neurons are within threshold distance (the nearer the higher weight). Those out of range should remain at weight =0. How should I implement this?

If I give a string like "-0.2+1/d" for the weights and the delays, and also make eceptor_type='excitatory'. Does this mean all negative weight will be ignored? If this is not the way, can I extract 'd' and use if statement? 

Also I have problem connecting to a single neuron in a population. I got this error 'pyNN.errors.ConnectionError: postsynaptic_neurons must be a Population, PopulationView or Assembly, not a <class 'pyNN.nest.simulator.ID'>' 

Is there a way to only project to a single neuron in a population? Many thanks.

Best regards,
Aries

**Comment by apdavison** [comment]
### Distance-dependent weights and delays

To assign a fixed weight of 0.2 for connections between neurons within 30 µm of each other and a weight of zero for all other connections, use
```python
weight="0.2 * (d < 30)"
```

To have a higher weight for nearer neighbours, you could do something like
```python
weight="0.1*(30 - d) * (d < 30)"
```

For `receptor_type='excitatory'`, all weights must be positive or zero, so negative weights will produce an error. For more complex weight rules, you can use a function rather than a string, e.g.:

```python
def weight_func(d):
    w = -0.2 + 1/(0.0001 + d)
    w[w < 0] = 0.0
    return w

...

weight=weight_func
```


### Connecting to a single neuron

To connect to a single neuron you have to connect to a population or population-view of size 1, e.g.
```python
prj = sim.Projection(population1, population2[0:1], ...)
```
rather than
```python
prj = sim.Projection(population1, population2[0], ...)
```
i.e. `population2[0]` gives a single neuron (represented by the `ID` class), `population2[0:1]` gives a view of the population that contains a single neuron.

**Comment by dsc1126** [comment]
Excellent! It works. Thank you.
